src/psyclone/tests/psyir/nodes/member_test.py

```diff
@@ -97,3 +96,50 @@ def test_member_equality():
 
     assert mem == mem2
     assert mem != mem3
+
+
+def test_member_lbound(fortran_writer):
+    ''' Tests for the lbound() method of Member. '''
+    one = nodes.Literal("1", symbols.INTEGER_TYPE)
+    two = nodes.Literal("2", symbols.INTEGER_TYPE)
+    # First, test when we don't have type information.
+    grid_type = symbols.DataTypeSymbol("grid_type", symbols.DeferredType())
+    sym = symbols.DataSymbol("grid_var", grid_type)
+    ref = nodes.StructureReference.create(sym, [("data", [one.copy()])])
+    lbnd = ref.member.lbound(0)
+    assert isinstance(lbnd, nodes.BinaryOperation)
+    out = fortran_writer(lbnd).lower()
+    assert out == "lbound(grid_var%data, 1)"
+    usym = symbols.DataSymbol("uvar", symbols.DeferredType())
+    ref = nodes.ArrayOfStructuresReference.create(
+        usym, [one.copy()],
+        [("map", [one.copy(), two.copy()]),
+         ("data", [one.copy()])])
+    lbnd = ref.member.member.lbound(0)
+    assert isinstance(lbnd, nodes.BinaryOperation)
+    out = fortran_writer(lbnd).lower()
+    assert out == "lbound(uvar(1)%map(1,2)%data, 1)"
+    # Second, test when we do have type information.
+    a2d = symbols.ArrayType(symbols.REAL_TYPE, [2, (2, 8)])
+    # Structure that contains "map" which is a 2D array.
+    stypedef = symbols.StructureType.create(
+        [("map", a2d, symbols.Symbol.Visibility.PUBLIC)])
+    stypedefsym = symbols.DataTypeSymbol("map_type", stypedef)
+    # Structure containing a structure of stypedef and an array of such
+    # structures.
+    stypedef2 = symbols.StructureType.create(
+        [("grid", stypedef, symbols.Symbol.Visibility.PUBLIC),
+         ("subgrids", symbols.ArrayType(stypedefsym, [3, (2, 6)]),
+          symbols.Symbol.Visibility.PUBLIC)])
+    ssym = symbols.DataSymbol("var", stypedef2)
```

Comment:
I find this test very hard to follow - would it be easier to understand and shorter if you would parse a Fortran expression (or even short program) instead?

Reply:
I don't think this is my code.